packages/material-ui/src/test-utils/describeConformance.js

```diff
@@ -107,11 +120,35 @@ function describeRef(element, getOptions) {
   });
 }
 
+/**
+ * Tests that the root component has the root class
+ *
+ * @param {React.ReactElement} element
+ * @param {() => ConformanceOptions} getOptions
+ */
+function testRootClass(element, getOptions) {
+  it('applies to root class to the root component if it has this class', () => {
+    const { classes, mount } = getOptions();
+    if (classes.root == null) {
+      return;
+    }
+
+    const wrapper = mount(element);
+
+    // we established that the root component the outermost previously. We immediately
+    // jump to the host component because some components pass the `root` class
+    // to the `classes` prop of the root component.
+    // https://github.com/mui-org/material-ui/blob/f9896bcd129a1209153106296b3d2487547ba205/packages/material-ui/src/OutlinedInput/OutlinedInput.js#L101
+    assert.strictEqual(findOutermostIntrinsic(wrapper).hasClass(classes.root), true);
+  });
+}
+
 const fullSuite = {
-  class: testClassName,
   componentProp: testComponentProp,
+  mergeClassName: testClassName,
   propsSpread: testPropsSpread,
```

Comment:
Should we refactor this test to use the same strategy as testClassName (findOutermostIntrinsic)? Maybe we can use if for `testComponentProp` too 🤔.

Reply:
I see I still failed to explain why props spread is not bound to host components in our docs. If we say we spread to `Paper` we have to test for that. Not just the outermost host.

Reply:
This makes sense, thanks 👍. I was looking for a potential simplification of the tests.